src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -581,3 +582,43 @@ def finalize(self) -> InheritedContext:
             return self
         tree = self._tree.renumber(-1)
         return InheritedContext(tree, self.found)
+
+
+T = TypeVar("T")
+
+
+class DependencyGraphWalker(abc.ABC, Generic[T]):
+
+    def __init__(self, graph: DependencyGraph, walked_paths: set[Path] | None, path_lookup: PathLookup):
+        self._graph = graph
+        self._walked_paths: set[Path] = set() if walked_paths is None else walked_paths
+        self._path_lookup = path_lookup
+
+    def walk(self) -> Iterable[T]:
+        for dependency in self._graph.root_dependencies:
+            root_path = dependency.path  # since it's a root
```

Comment:
what does this comment explain?

Reply:
that the path from which to collect python globals is the path of this dependency

Reply:
can we add that as a code comment then? :)

Reply:
done